internal/docker/client.go

```diff
@@ -109,6 +110,15 @@ type Client interface {
 	ContainersPrune(ctx context.Context, pruneFilters filters.Args) (types.ContainersPruneReport, error)
 }
 
+// Add-on interface for a client that manages multiple clients transparently.
+type CompositeClient interface {
+	Client
+	DefaultLocalClient() Client
+	DefaultClusterClient() Client
+	ClientFor(cluster v1alpha1.Cluster) Client
```

Comment:
Leaving this as a hint for multi-cluster. I can envision other possible methods like:
```go
AddClient(client Client, cluster v1alpha1.Cluster)
```